src/main/java/org/reficio/p2/P2Mojo.java

```diff
@@ -276,8 +276,13 @@ private void processTransitiveArtifacts(Multimap<P2Artifact, ResolvedArtifact> r
         for (P2Artifact p2Artifact : artifacts) {
             for (ResolvedArtifact resolvedArtifact : resolvedArtifacts.get(p2Artifact)) {
                 if (!resolvedArtifact.isRoot()) {
-                    if (bundledArtifacts.add(resolvedArtifact.getArtifact())) {
-                        bundleArtifact(p2Artifact, resolvedArtifact);
+                    if (!bundledArtifacts.contains(resolvedArtifact.getArtifact())) {
+                        try {
+                            bundleArtifact(p2Artifact, resolvedArtifact);
+                            bundledArtifacts.add(resolvedArtifact.getArtifact());
+                        } catch (final RuntimeException ex) {
+                            log.warn(String.format("Skip artifact=[%s]: %s", p2Artifact.getId(), ex.getMessage()));
+                        }
                     } else {
                         log.debug(String.format("Not bundling transitive dependency since it has already been bundled [%s]", resolvedArtifact.getArtifact()));
```

Comment:
Looks good, but I would invert it, not to break compatibility.
Right now it fails - maybe we could add -lenient flag that would cause the build to continue in such cases.